### PR TITLE
refactor: use updated config provider function for isCommonConfigReady

### PR DESCRIPTION
**Pull request by ejlee3**
If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) - docs need to be added for common config
  <link to docs PR>

## Testing Instructions
**Non-secure mode testing**
1. set the environment variable for non-secure testing `export EDGEX_SECURITY_SECRET_STORE=false`
2. run the latest compose builder `make run no-secty` to start core services
3. stop core data
4. remove the core data configuration in consul
5. modify local go-mod in edgex-go to use bootstrap from [this branch](https://github.com/ejlee3/go-mod-bootstrap/tree/update-config-ready) and config from [this branch](https://github.com/ejlee3/go-mod-bootstrap/tree/update-config-ready) using `replace` statement
7. build common config and core data `make data`
9. run core-data locally and ensure that the configuration loads `cd cmd/core-data` and `./core-data -cp`
10. Verify that the core data configuration was loaded into consul
11. Modify the writable log level and ensure the terminal reflects the change

**Secure mode testing**
1. set the environment variable for non-secure testing `export EDGEX_SECURITY_SECRET_STORE=true`
2. run the latest compose builder `make run` to start core services
3. stop core data
4. remove the core data configuration in consul
5. modify local go-mod in edgex-go to use bootstrap from [this branch](https://github.com/ejlee3/go-mod-bootstrap/tree/update-config-ready) and config from [this branch](https://github.com/ejlee3/go-mod-bootstrap/tree/update-config-ready) using `replace` statement
7. build common config and core data `make data`
9. run core-data locally and ensure that the configuration loads `cd cmd/core-data` and `sudo ./core-data -cp`
10. Verify that the core data configuration was loaded into consul
11. Modify the writable log level and ensure the terminal reflects the change

### CODE REVIEW

File: bootstrap/config/config.go [hunk 1]
```diff
@@ -228,19 +228,16 @@ func (cp *Processor) loadCommonConfig(
...
-	if err := cp.waitForCommonConfig(commonConfigCheckClient); err != nil {
-		return err
-	}
 	// load the all services section of the common config
 	commonConfigClient, err := createProvider(cp.lc, common.CoreCommonConfigServiceKey+allServicesKey, configStem, getAccessToken, configProviderInfo.ServiceConfig())
 	if err != nil {
 		return fmt.Errorf("failed to create provider for %s: %s", allServicesKey, err.Error())
 	}
+	// build the path for the common configuration ready value
+	commonConfigPath := fmt.Sprintf("%s/%s", configStem, common.CoreCommonConfigServiceKey)
```
Comment (lenny-intel) [inline bootstrap/config/config.go:237]: Why not also include `config.CommonConfigDone` here so it doesn't have to be appended later?

File: bootstrap/config/config.go [hunk 2]
```diff
@@ -668,7 +665,7 @@ func (cp *Processor) waitForCommonConfig(configClient configuration.Client) erro
 	isConfigReady := false
 	isCommonConfigReady := false
 	for cp.startupTimer.HasNotElapsed() {
-		commonConfigReady, err := configClient.GetConfigurationValue(config.CommonConfigDone)
+		commonConfigReady, err := configClient.GetConfigurationValueByFullPath(fmt.Sprintf("%s/%s", configReadyPath, config.CommonConfigDone))
```
Comment (lenny-intel) [inline bootstrap/config/config.go:668]: This is appending `config.CommonConfigDone` each iteration.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: bootstrap/config/config.go (hunk 1 above)
Comment (ejlee3) [reply bootstrap/config/config.go:237]: ok. can do - was debating which way to go.

File: bootstrap/config/config.go (hunk 2 above)
Comment (ejlee3) [reply bootstrap/config/config.go:668]: fixed

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**